atf_core/src/atf_core/recorder.py

```diff
@@ -189,6 +213,13 @@ def create_subscriber(self, topic):
                 msg_class = TFMessage
             else:
                 msg_class, _, _ = rostopic.get_topic_class(topic)
+            rospy.logdebug("received message class for topic %s: %s"%(topic, str(msg_class)))
+        except Exception as e:
+            msg = "Error while requesting message class for %s: %s."%(topic, e)
+            rospy.logerr(msg)
+            #raise ATFRecorderError(msg)
+            return None
+        try:
             subscriber = rospy.Subscriber(topic, msg_class, self.global_topic_callback, callback_args=topic)
```

Comment:
this still fails multiple times because `msg_class` is `None`

related:
why is `create_subscriber` called in a Timer callback?